### ISSUE TITLE
Pyright incorrectly claims an awaitable isn't awaitable when `__await__` is a descriptor

### ISSUE THREAD

**Issue by mikeshardmind** [microsoft/pyright#9739]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**

When having `__await__` implemented as a descriptor, pyright incorrectly claims the object in question isn't awaitable, while correctly determining that it has the correct type for `.__await__()` to be awaitable.

This includes when the descriptor is implemented such that that access from the type results in an equivalent access such that `type(x).__await__(x)` is equivalent to `x.__await__()`, and that the descriptor works for other delegated access other than await as intended.

see that delegation of the internal future's `.set_result`, `.done`, and `.cancelled` methods are all properly understood, but not `.__await__`

This works at runtime.

**Code or Screenshots**

Full code in question [pyright playground](https://pyright-play.net/?code=MQAjGEHsAcE8CcCWBzAFgFxAJgAy4LTTwCmAzsQHaYCyiAxqgIbEA2IAEoyywFDB9gQAGXqVyAExABXCuOLwQ6VMRABBaIwYqRdMcQA0IAGrzSiSBWwA6HCAAUSlQCIdepwEoA3ALCxIUkABbRlgQCkhMKXJFVERSEAAzRBYVYgAPXWhMREs6SEDoFkRGCl0QAHdEJRjtUQpyKx8QAE1-IJCQSAAjdEYckEYQPLhOhJrhOujGdD4mwRAMdGgALgB6VfLNq0YNLStIeGRVot16slWhAElwAFEAOQBlG-wsG1nQMABVChTSeJIAI5SRAkSRdUI7Qr0RhdFIgFiMcqdBSMZAkYiSdCQED9cpIdA5ZCGUiQBLocqMEhNcRxdBILpSdAY6SyeTjVxnHHxGmkOmIBlMyQWAaWJyqB4gS4PJwgABC4ql%2BiaAHVLgAVdgAeU%2BapAytUACUDao7mrLjcJZqDSBwJq7gARdWXO2WgBiajuzRAAGlLg7DMQqsoFOkiGR4gccQUihjGh8QA9iCpHBNTtEEpGU6RoMQ6IgknR4SVkFJUSpkJAAG7yCiEkA5%2BCBOJmCzxEriJpFJu9AmtllyBQpjkNHijhLwfIgAD6U4SjKkJBnUegB0wJXCPfM9VHiAKq4GpFgpXMPF3K-gmDyVHS6ErlNIp73F4WxB2AMf58wShIjBpFGQPDjpOeTcLmvb1NsXSFme%2B7gFwCKwgYIAAOKUPI0wHIBE6BEMFhMmk6BFF0y77le%2BHoMEFBlvAWGTjA6FYgoMHPtMdLIMQ6BMjRH77ugsDQHWjDxDMPAzvBS4ALz2E4AAKSAHFUsCJsE0CoAcxBOIYThEOY%2BKwFOZE3h4o5TsgLDdFwU7mXQADWIBSd%2Br5-sgVhCJAtl2O4Jk6QpfHLLh14EXe8CkFYUCBegRiUgA2jk6AALr2QF5HBaF4XkVF8B2E4U4%2BXpmkgHICSMFILDoBJOBeaOdAIn89YTtA0VqoYBrxcsPDzEV05TjkVQznY5AsAkhgJOx6D%2BXB3Awik0VNfFLXxe4ID4AAfCAdwWMQ7XzPMg0JFYo0cUlh0iZ1xBjDOdDiVOA2sAkS2rSABr%2BVYr2jmdF1TuQ6D9Xthg5LyJS6P5uoAD7rZthiQOUFDyP5fE5nND1regVh3MQ1bwNtO1gPAfTRKonH0oyxA3PAE7cfM%2BaKFYarNDJNxTuA7A3OAvp3Mh7UdTjIAAAKo1W8jmb%2B3M411M5jb9d1Q10ABWIMy7LCNbYo-HEHNIDgxtsPI09L1vaLO38-smPCx2POFed3WSzdf2dHL-na0h3RK2r8NqxrWubbrk0ITN0Wo6oFCwPNT1tSAr2NIb4umRxUtDYrIOaxDsOK8r8NWEHsDuNjOPUy7XIp1thsWyQ6ALpYe0HWNJeCGXFcgFXJ12C7VU8DVQnxHJumKcqfRcbn3WkOZ6CkJJ0lzuXJAFU4BziMZ0dWzOvU-bb0v1T3fk4lQhij-5CTC%2BgI3ziQ-lCUeeaQFYron%2BrTuh6sutO4Pu13fs8CSFJdh5Ypu%2BkF5Fsm632Orfd6ggrqlFYCkcQ-kiAwADlYRMCcbTwWmuraKod76JSknA6AdhWKHA4lxbKk8FzECsBA3QoF57uAAYIcQm1YENQQUg4aKCpqIVmpgza8VsH1RgPgomY1iFOFISQKwDDYYeDoWAMSFI%2BpTiYfA1GrDDC%2BzQVwwwqFYZ40YggrOhgnaGJ4bwpKuDBFsSIfIEht8rByP7jOaRhtvpThIKQUq41%2BGNRUevdRnDor32MbDUxOCGoWMIZxaxojbEuLcR4pxi9PplXjgkDOqjOiOCxvbWWYFdZdEgJAFgL9BDUw3FyAGvRIEt0yYYbuvlYB9yqPIHOtccb13gJYDa6BLjRmIIESggpWkgHaZXN%2Bc8QAAB4MnBnfh2Q2M4khURYOPNU8ApDEDAWAGOK8vpSCgrVMeN0ar-2WijNGGM4ZDMCKQZASUnD2gsAAckwO4-ZncYhxCcEMvGcQVAGhkASfpZMKZ2GucgNuvNDIEUotRHgXUf58QMnhG8395J6X8nFQwj9TkoTQrog4ATIZFyCcQNqhsnAUsTF%2BZQG96mJEjPBEAdS9LKR2GpEg0hoj9CUHEZKN4o7zBkpSRg-SuIPnmPgSVUrpWGwRbADFVAhlqhpXKxQ2IojkOENDNkd4WDrLbBy0kAwFgoGDLSvSjR5gUq%2BfMLENlKBJVymixSVhvqos3tnQ2dJ5VDNgIgVgkgnaG0WfBH1FsnUeqsG4jiDhID2ooG3KcRUSplQzlnJKLswKeVHO3A5TLnV8VZapdSg9rWqATH0tl6kKhBnNYpfAXQhLMkYECOIVQtzIgHISAVgg5JVkQHINsjdK3Fo5ZUaoZgmwIgUOQSiBI6BtkXRfcwiCR3soMIbAUnLBKWBkAOFgfr-x1r4lYY9oReWkCYKCGtPLLCDChTMeYML2LwEefEcyyBoRsFIMMJCW6UwPqGCUEAXQVCw15MyUgjAEisFgD2sAblyhslVbq-VAwOWDFiGgNkH6v1ns2Uy4VorTCG2lWR-AhtUMqzikqmlK9ihsCo6McYcUaxcFwgC%2BQlqebKrA1IQIoGFBGtvcgYSTBLzAavHQBcJAqAHoGHQVtHKqjwZADcNIIrChkFI%2BRij8wVoGeHThKSzLFJFvXXYAAjDIkABm1rjtQGegy6A0hWcqsUnG59Sg3sc7OjzFsI4GytRSgjVkCkrAPMuq%2Bqgui8jxnQdANxqxUDcjAZOTskpBsSd1HZa9kFUYVZgKS1mcXPyGdTJjUycD%2BafTcu55nq30fY0xwIURMCgds1JHANqAu43xioKKerSbkwOKCm5NnX5DTsfIsV-kii8miqZvijSuKJS9rDTLm0hlVynAV7eRWQBUYI%2BLSW5kBF7V1l5y%2BmdYt0k0Il5L6BUsrCGWd6ASUrsrslmsigtZ-xhYEVVC21MdtvcLmV3rYAHPTjMhZZZ1kbI1Z5iDt%2BAP3u8oh5DwBqOwdSTe%2BVsYYPeVlNB%2BFpHYAwXHScAAbz2gAQngAAX0LvkvdarjU0gSDB2TmALlUHhOFnrvWfnRH%2BVQXcw2QVgom3XDiDd8fZZnCQIgKTdZxf824pK7iGyeTsa44gKuboy7ADePGdyEcYhlCjqbM4LfiE8iAVg6YnAyDt4YArtPUdUcZ0Lna1v9pThm6YcnjuCJm6kgkGnpvGCM8MEHkKyxqcpAoLdG38f-4%2B%2B%2BXLjpiQnATOp246Klnlj4Es-FZn0VqfR8Z-FFaXzFdhdshiVP90cX5MKf50A1A2uFWxPe-IGgkAkksI2iQnRLCqscJYV0lxXSaiGaAblNKrrRCNekWkdZ08L7CBEEAJZhXXmZFiEDKhdz9JpNMWDAw7zJDQVnqeoybdMYkl15E9gSiwDsGU8oFCgZQOb0tBmAoEiP0C3tNv3KYG-p5LQkDoIFdpbJ9MwNePAKrqVowkMj5I6nKlYGNNmhbIiP3MOjblOJoEpsQO6pALAWAPATHMwGkAoi3oYAAFSSI-RXSnzZK5JoGwz%2BY7b64pBNrZqGw0FLwkGKbAgkCMFnqFaOrJoeJ5IFJFIE5nqFxJrnQprjRDLzCqpSQRr1I4ExrG44hjAk6o6N72r24tJY47ZMb4DFZaGy4P4gCrLrIEZUxjA7bp7g7oHWGo5eFSQYJuGCC45EEB6nbhZCEWyTxJRvYUI-hMizi3x4E8zhBIh47hZWCArkFGHx5JRLYNIQFZRyqGCpHHzoBUE7TKBvhWBVHQDQBRCoBgGB6FGkBx6FEVHzCpKRbHhXw3xTx3w8IfaHg9HXxJFGGowkDViWTKx2AB4kEzYzjjFRoXLTFqx2Dx564EEKIdGCDeoh5bGYAJAOHUEzYVCFHHEgCgCu6ID2rGrhCNjsazpVokBx4qCSLPLDJ9KCydALiDTVjiqQ6gBIZhBJiYjYjBBxR9CWApi1HbzBTFBUAAk4zpCZBrjDHXZwSQLULAoHAh7UyTwSKbQO7tg76HGMi-5YnQKeQh6TYB5MYADU9hWOIuGyQyIa3AYakOO2wQsAoGzR9qyRbS2elgLhrJH004PJfJFIApF23BxcFs5QsQcINhXA6ytmIAtgJJnhLRIesMBEzRTSCgUktRAINRr4dR52fhLROxecJhu%2BX%2BN4BpXEhJsMDukYepP0GxlC-%2BlhVhWOggKpQ2y0TJ-pYAHpTpXGsSZAHidgTsNpFQSpKg2phppAIeoANUr4FAUg72EJVAUJdYsJOQ8JJQj6gJzhsQ8QlQ3A4%2B8mc4HJjuBQfE4wsJiplAFxoAjKIm-wlavUR6S%2BKgsJ0GXEZxNkdYvQpANk8QOQ7ZAw%2BS1Yw6CWHalIKgjKki5CFx4ZuRxp5ppptRK4eCyZYqRh7hpJ9gm5hRLp2Rb%2B55hpFJVCVJtCNJggJpZpkIDRTR6eJRjp8ex5PMXQP4Nkx2ohJAAh5ALeT8PhPMgZapjJIAlm22qOkpxA-J2RPAQAA) with an inserted reveal type and comparison with something that clearly is detected as awaitable with the same detected type for `.__await__`

**Comment by mikeshardmind** [comment]
@erictraut Seems compatible with awaitable to me, if it isn't, that seems to be an issue in how compatibility is defined.

Here's the entire stub in the typeshed for awaitable:

https://github.com/python/typeshed/blob/ce521e8c76eec84828b30353b163fcff8e054bbd/stdlib/typing.pyi#L507-L510

```py
@runtime_checkable
class Awaitable(Protocol[_T_co]):
    @abstractmethod
    def __await__(self) -> Generator[Any, Any, _T_co]: ...
```
